tests/phpunit/Blt/TestDrupalTestCommandTest.php

```diff
@@ -0,0 +1,189 @@
+  }
+
+  /**
+   * Test Drupal's Simpletest type with run-tests.sh.
+   */
+  public function testDrupalRunTestsSimpletestTests() {
+    list($status_code, $output, $config) = $this->blt("tests:drupal:run", [
+      "--define" => [
+        "tests.drupal.test-runner=run-tests",
+        "tests.run-tests.0.color=true",
+        "tests.run-tests.0.concurrency=1",
+        "tests.run-tests.0.repeat=1",
+        "tests.run-tests.0.tests.0=user",
+        "tests.run-tests.0.types.0=Simpletest",
+        "tests.run-tests.0.sqlite=$this->sqlite",
+        "tests.run-tests.0.url=$this->url",
+      ],
+    ]);
+    $results = file_get_contents($this->sandboxInstance . '/reports/phpunit/Drupal_user_Tests_RestRegisterUserTest.xml');
+    $this->assertNotContains('failure type="failure"', $results);
+  }
+
+  /**
+   * Test Drupal's PHPUnit-Unit type with run-tests.sh.
+   */
+  public function testDrupalRunTestsUnitTests() {
+    list($status_code, $output, $config) = $this->blt("tests:drupal:run", [
+      "--define" => [
+        "tests.drupal.test-runner=run-tests",
+        "tests.run-tests.0.color=true",
+        "tests.run-tests.0.concurrency=2",
+        "tests.run-tests.0.repeat=1",
+        "tests.run-tests.0.tests.0=action",
+        "tests.run-tests.0.types.0=PHPUnit-Unit",
+        "tests.run-tests.0.sqlite=$this->sqlite",
+        "tests.run-tests.0.url=$this->url",
+      ],
+    ]);
+    $results = file_get_contents($this->sandboxInstance . '/reports/phpunit/Drupal_Tests_action_Unit_Menu_ActionLocalTasksTest.xml');
+    $this->assertNotContains('failure type="failure"', $results);
+  }
+
+  /**
+   * Test Drupal's PHPUnit-Kernel type with run-tests.sh.
+   */
+  public function testDrupalRunTestsKernelTests() {
+    list($status_code, $output, $config) = $this->blt("tests:drupal:run", [
+      "--define" => [
+        "tests.drupal.test-runner=run-tests",
+        "tests.run-tests.0.color=true",
+        "tests.run-tests.0.concurrency=2",
+        "tests.run-tests.0.repeat=1",
+        "tests.run-tests.0.tests.0=action",
+        "tests.run-tests.0.types.0=PHPUnit-Kernel",
+        "tests.run-tests.0.sqlite=$this->sqlite",
+        "tests.run-tests.0.url=$this->url",
+      ],
+    ]);
+    $results = file_get_contents($this->sandboxInstance . '/reports/phpunit/Drupal_Tests_action_Kernel_Migrate_d7_MigrateActionsTest.xml');
+    $this->assertNotContains('failure type="failure"', $results);
+    $results = file_get_contents($this->sandboxInstance . '/reports/phpunit/Drupal_Tests_action_Kernel_Plugin_Action_EmailActionTest.xml');
+    $this->assertNotContains('failure type="failure"', $results);
+    $results = file_get_contents($this->sandboxInstance . '/reports/phpunit/Drupal_Tests_action_Kernel_Plugin_migrate_source_ActionTest.xml');
+    $this->assertNotContains('failure type="failure"', $results);
+  }
+
+  /**
+   * Test Drupal's PHPUnit-Functional type with run-tests.sh.
+   */
+  public function testDrupalRunTestsFunctionalTests() {}
```

Comment:
Functional tests will fail unless be used the same version of `behat/mink` as core. See: https://www.drupal.org/project/drupal/issues/2948700